Model/SubscriptionManagement.php

```diff
@@ -107,11 +127,44 @@ public function cancelSubscription($subscriptionId)
             }
 
             $this->subscriptionRepository->save($subscription);
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
             $this->logger->error($e->getMessage());
             throw new LocalizedException(__("Subscription couldn't be canceled"));
         }
 
         return __('Subscription has been canceled correctly');
     }
+
+    /**
+     * Change assigned card for subscription
+     *
+     * @param string $subscriptionId
+     * @param string $cardId
+     *
+     * @return bool
+     *
+     * @throws LocalizedException
+     */
+    public function changeSubscription(string $subscriptionId, string $cardId): bool
+    {
+        $paymentToken = $this->paymentTokenRepository->getById((int)$cardId);
+        $customerId = $this->userContext->getUserId();
+
+        try {
+            $subscription = $this->subscriptionRepository->get((int)$subscriptionId);
+
+            if (!$paymentToken || (int)$paymentToken->getCustomerId() !== $customerId ||
+                (int)$subscription->getCustomerId() !== $customerId
+            ) {
+                throw new Exception();
+            }
+
+            $subscription->setSelectedToken((int)$paymentToken->getEntityId());
+            $this->subscriptionRepository->save($subscription);
+
+            return true;
+        } catch (Exception $e) {
+            throw new LocalizedException(__('Unable to change card'));
+        }
```

Comment:
Optional: I would recommend restructuring this code for improved readability. 
- You could move the payment token load to a new private function within the this class that is responsible for loading a token, and then validating that the card has correct user id (throwing localizedException if it doesn't).
- After the card has been validated and returned to this function, you can validate the subscription by comparing it against card and throwing a localized exception. 
- Finally when card and subscription are valid you could save them in a separate private function with it's own try catch.

Why do it like this? 
Did you notice that now your code is using a very low level exception to interrupt itself should the validation not pass. Catching this low level exception also causes you to catch other errors such as database errors that could rise from subscription repository. I believe it would be beneficial for you to allow catching these repository exceptions in a different try catch in a dedicated save function in case you want logging. For your own code to directly throw LocalizedExceptions when a validation does not pass. For readability i would recommend to keep try catches as small as you can make them.